### DISCUSSION TITLE
How to unit test that watch-queries never emit event if query-conditions not met?

### DISCUSSION THREAD

**Discussion by NikoBoerger** [simolus3/drift#1504] [Q&A] [answered]
Hi,

I've got a question about the Streams created by Watch-Queries. I want to write UnitTests that check that the Stream never emits anything if some query-conditions aren't met. Since the stream is never closed though, you can't use code like this:

```Dart
    test("should not emit anything if no unfinished match found", () async {
      // given: no unfinished match in db

      // when
      Stream<UnfinishedMatchModel?> unfinishedMatchStream = _matchControlDao.watchLastUnfinishedMatch();

      // then
      expectLater(unfinishedMatchStream, neverEmits(anything));
    });
```

This tests runs into the timeout, because the stream is never closed. Is there a way to close the stream manually or do you know another way to test this?

Thanks!

**Comment by simolus3** [comment, marked as answer]
By design, query streams live as long as your database. So if you call `.close()` on the database, the streams would close too. So you could do a bunch of updates to the database in the test and call '.close()` in the end. There's not much more than can be done about this apart from that unfortunately.

**Comment by NikoBoerger** [comment]
Awesome, exactly what I needed to know. Thank you so much!
